samples/matter/lock/src/app_task.cpp

```diff
@@ -499,6 +494,18 @@ void AppTask::ChipEventHandler(const ChipDeviceEvent *event, intptr_t /* arg */)
 		}
 #endif
 		break;
+#elif defined(CONFIG_CHIP_WIFI)
+	case DeviceEventType::kWiFiConnectivityChange:
+		sIsNetworkProvisioned = ConnectivityMgr().IsWiFiStationProvisioned();
+		sIsNetworkEnabled = ConnectivityMgr().IsWiFiStationEnabled();
+#if CONFIG_CHIP_OTA_REQUESTOR
```

Comment:
I'm not sure I remember correctly. @Damian-Nordic correct me if I'm wrong, but I think it was to make sure that networking is enabled before initializing OTA requestor to avoid NotifyUpdateApplied command timeout.

Reply:
> Isn't there a chance that the connection is reestablished and we may initialize the requestor more than once?

No, because the InitBasicOTARequestor() will not allow to that. It returns if valid instance is already created.

Reply:
Ah here are some related changes pending:
https://github.com/nrfconnect/sdk-nrf/pull/8445/files

Reply:
Good point, I show align to that.